tests/common/snappi_tests/snappi_fixtures.py

```diff
@@ -612,62 +611,69 @@ def tgen_ports(duthost, conn_graph_facts, fanout_graph_facts):  # noqa: F811
             port['speed'] = speed_type.get(str(port_speed), port['speed'])
             peer_port = port['peer_port']
             int_addrs = list(config_facts['INTERFACE'][peer_port].keys())
-            port['speed'] = speed_type.get(str(port_speed), port['speed'])
-
-            ipv4_entry = next((a for a in int_addrs if '.' in a), None)
-            if ipv4_entry:
-                port['peer_ip'], port['prefix'] = ipv4_entry.split('/')
-                port['ip'] = get_addrs_in_subnet(ipv4_entry, 2)[1]
-                snappi_ports_all.append(port)
-            else:
-                un_ipv4.append(port)
-
-            ipv6_entry = next((a for a in int_addrs if ':' in a), None)
-            if ipv6_entry:
-                port['peer_ipv6'], port['ipv6_prefix'] = ipv6_entry.split('/')
-                port['ipv6'] = get_ipv6_addrs_in_subnet(ipv6_entry, 2)[1]
-                snappi_ports_all.append(port)
-            else:
-                un_ipv6.append(port)
-
-        if un_ipv4:
-            snappi_ports_all.extend(pre_configure_dut_interface(duthost, un_ipv4, type='ipv4'))
-        if un_ipv6:
-            snappi_ports_all.extend(pre_configure_dut_interface(duthost, un_ipv6, type='ipv6'))
-    return snappi_ports_all
+            for ipver, addr_type in (("ipv4", "IPv4"), ("ipv6", "IPv6")):
+                entry = next((a for a in int_addrs if (":" in a) == (ipver == "ipv6")), None)
+                if ipver == "ipv4":
+                    dut_list, tgen_list, mask = dutIps, tgenIps, prefix_length
+                    peer_ip_key, prefix_key, ip_key = "peer_ip", "prefix", "ip"
+                else:
+                    dut_list, tgen_list, mask = dutv6Ips, tgenv6Ips, v6_prefix_length
+                    peer_ip_key, prefix_key, ip_key = "peer_ipv6", "ipv6_prefix", "ipv6"
+                if entry:
+                    # Already configured on DUT
+                    port[peer_ip_key], port[prefix_key] = entry.split("/")
+                    port[ip_key] = get_addrs_in_subnet(entry, 1, exclude_ips=[entry.split("/")[0]])[0]
+                else:
+                    # Assign and configure new IPs
+                    port[peer_ip_key] = dut_list[port_id]
+                    port[prefix_key] = mask
+                    port[ip_key] = tgen_list[port_id]
+                    try:
+                        logger.info(
+                            f"Pre-configuring {addr_type}: {duthost.hostname} "
+                            f"port {peer_port} -> {dut_list[port_id]}/{mask}"
+                        )
+                        duthost.command(
+                            f"sudo config interface ip add {peer_port} {dut_list[port_id]}/{mask}"
+                        )
+                    except Exception as e:
+                        pytest.fail(
+                            f"Unable to configure {addr_type} on {peer_port}: {e}",
+                            pytrace=False,
+                        )
+    return snappi_ports
```

Comment:
Old code accumulated ports from multiple fanouts into `snappi_ports_all`. New code only returns `snappi_ports` from the last iteration of the fanout loop. Is this intentional?